### PR TITLE
CAMEL-14738: Create a camel-platform-http-vertx component

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

**Comment by lburgazzoli** [comment]
@davsclaus @oscerd @ppalaga mind having a look ?

**Comment by oscerd** [comment]
LGTM, I don't think we need a starter for this right?

**Comment by lburgazzoli** [comment]
I'd say no, we probably need a platform-http engine for spring boot but that's a separate issue

**Comment by lburgazzoli** [comment]
but I can work on a starter for this new component once merged

File: components/camel-platform-http-vertx/pom.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    <parent>
+        <groupId>org.apache.camel</groupId>
+        <artifactId>components</artifactId>
+        <version>3.2.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>camel-platform-http-vertx</artifactId>
+    <packaging>jar</packaging>
+    <name>Camel :: Platform HTTP :: Vert.x</name>
+    <description>Implementation of the Platform Http Engine based on Vert.x Web</description>
```
Comment (davsclaus) [inline components/camel-platform-http-vertx/pom.xml]: HTTP

File: components/camel-platform-http-vertx/src/main/java/org/apache/camel/component/platform/http/vertx/VertxPlatformHttpServer.java [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class implement a basic Vert.x Web based server that can be used by the {@link VertxPlatformHttpEngine} on
+ * platforms that do not provide Vert.x based http services.
+ * <p/>
+ * Example:
+ * <pre>{@code
+ *  final int port = AvailablePortFinder.getNextAvailable();
```
Comment (davsclaus) [inline components/camel-platform-http-vertx/src/main/java/org/apache/camel/component/platform/http/vertx/VertxPlatformHttpServer.java]: Maybe add this to the adoc documentation file instead, I think its better for user documentation

File: components/camel-platform-http-vertx/src/generated/resources/platform-http-vertx.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "other": {
+    "kind": "other",
```
Comment (ppalaga) [inline components/camel-platform-http-vertx/src/generated/resources/platform-http-vertx.json]: Shouldn't this be a component?

File: components/camel-platform-http-vertx/src/main/doc/platform-http-vertx.adoc [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+== Auto detection from classpath
+
+To use this implementation all you need to do is to add the `camel-platform-http-vertx` dependency to the classpath,
+and the platform http component should auto-detect this.
+
+== VertxPlatformHttpServer
+
+In addition to the implementation of the `PlatformHttp` SPI based on Vert.x, this module provides a Vert.x based HTTP
+server compatible with the `VertxPlatformHttpEngine`:
```
Comment (ppalaga) [inline components/camel-platform-http-vertx/src/main/doc/platform-http-vertx.adoc]: Nice doc text!

File: components/camel-platform-http-vertx/src/generated/resources/platform-http-vertx.json (hunk 3 above)
Comment (lburgazzoli) [reply components/camel-platform-http-vertx/src/generated/resources/platform-http-vertx.json]: it is not a component, i.e. you can't use it in a `from(...)`

Comment (ppalaga) [reply components/camel-platform-http-vertx/src/generated/resources/platform-http-vertx.json]: Thanks for the explanation!

**Pull request merged**